### PR TITLE
can observe a entity multiple time

### PR DESCRIPTION
**Pull request by siba-tunerlabs**

### CODE REVIEW

File: src/components/observation-entity-listing/observation-entity-listing.html [hunk 1]
```diff
@@ -1,51 +1,58 @@
...
+                (click)="checkSubmission(entity,i,k)">
                 {{entity.name}}
+              </div> -->
+  <!-- 
+                <span float-right *ngIf="entity?.submissionStatus === 'inprogress'"
+                  class="_inProgress _flex-box">
+                  <ion-icon name="contrast" class="_progressIcon" margin-left></ion-icon>
+                </span> -->
+                <ion-icon *ngIf="entity?.submissionStatus == 'completed'" name="cloud-done" class="completed" margin-left></ion-icon>
+                <ion-icon *ngIf="entityList[0].status !== 'completed' && entity?.submissionStatus !== 'completed'" name="trash" class="downloadIcon" (click)="removeEntity(i,k)" margin-left></ion-icon>
```
Comment (kiranharidas187) [inline src/components/observation-entity-listing/observation-entity-listing.html]: enable only if the submissions count is 0

File: src/components/observation-entity-listing/observation-entity-listing.ts [hunk 2]
```diff
@@ -54,71 +58,114 @@ export class ObservationEntityListingComponent {
...
+      });
+    }
+  }
+
+  goToEcm( name, entityindex, observationIndex ,submissionId) {
+    // console.log("go to ecm called");
     let heading = name;
+    // this.observationProvider.getSubmission(this.selectedObservationIndex ,entityindex, 1 , this.observationList,'createdObservationList').then(success => {
+    // //   console.log(JSON.stringify(success));
+    this.localStorage.getLocalStorage('createdObservationList').then(success => {
```
Comment (kiranharidas187) [inline src/components/observation-entity-listing/observation-entity-listing.ts]: check if required

File: src/components/observation-entity-listing/observation-entity-listing.ts [hunk 3]
```diff
@@ -127,99 +174,102 @@ this.localStorage.getLocalStorage(this.utils.getAssessmentLocalStorageKey(submis
...
         }
-      })
+        entityList.forEach(element => {
+          payload.data.push(element._id);
+        });
+        // this.utils.startLoader();
+        this.apiProviders.httpPost(AppConfigs.cro.mapEntityToObservation + this.entityList[params[0]]._id, payload, success => {
+          // this.utils.stopLoader();
+
+          entityList.forEach(entity => {
```
Comment (kiranharidas187) [inline src/components/observation-entity-listing/observation-entity-listing.ts]: remove if not required

File: src/pages/observation-details/observation-details.ts [hunk 4]
```diff
@@ -19,65 +20,194 @@ export class ObservationDetailsPage {
...
+    private observationService:ObservationServiceProvider,
     private apiProvider: ApiProvider,
     private localStorage: LocalStorageProvider,
+    private observationProvider : ObservationServiceProvider,
     private events: Events) {
 
-      this.events.subscribe('observationLocalstorageUpdated', success => {
-        this.getLocalStorageData();
-      })
+    this.events.subscribe('observationLocalstorageUpdated', success => {
```
Comment (kiranharidas187) [inline src/pages/observation-details/observation-details.ts:39]: Remove if not required

File: src/pages/observation-details/observation-details.ts [hunk 5]
```diff
@@ -114,43 +244,44 @@ export class ObservationDetailsPage {
...
       this.goToEcm(this.navParams.get('selectedObservationIndex'), event, program)
     }).catch(error => {
 
     })
   }
-  goToSubmissionListPage(observationIndex,entityIndex) {
-    this.navCtrl.push(SubmissionListPage , {observationIndex : observationIndex , entityIndex : entityIndex , selectedObservationIndex :this.navParams.get('selectedObservationIndex')}  )
+
+  goToSubmissionListPage(observationIndex, entityIndex) {
+    this.navCtrl.push(SubmissionListPage, { observationIndex: observationIndex, entityIndex: entityIndex, selectedObservationIndex: this.navParams.get('selectedObservationIndex') })
```
Comment (kiranharidas187) [inline src/pages/observation-details/observation-details.ts]: recheck and remove unwanted variabled

File: src/pages/observation-details/observation-details.ts [hunk 6]
```diff
@@ -114,43 +244,44 @@ export class ObservationDetailsPage {
...
   }
 
-  goToEcm(observationIndex, event, program) { 
+  goToEcm(observationIndex, event, program) {
```
Comment (kiranharidas187) [inline src/pages/observation-details/observation-details.ts]: If needed add it to service

File: src/providers/observation-service/observation-service.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  },error=>{
+    event ? event.complete() : this.utils.stopLoader();
+    reject();
+  });
+    });
+    
+  }
+
+
+  getObservationList(){
```
Comment (kiranharidas187) [inline src/providers/observation-service/observation-service.ts]: remove

File: src/components/observation-entity-listing/observation-entity-listing.html (hunk 1 above)
Comment (siba-tunerlabs) [reply src/components/observation-entity-listing/observation-entity-listing.html]: DONE

**Pull request merged**